app/routes.py

```diff
@@ -1,2 +1,232 @@
+                
+            }
+
+        query_headers = {
+            "authorization": f"Bearer {key}"
+        }
+        response = requests.post(path, params=query_params, headers=query_headers )
+        response_body = response.json()
+
+        print(response)
+        
+        
+        db.session.commit() 
+        
+        return jsonify({"task": task.to_json()}),200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def task_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+    elif task:
+        task.completed_at = None
+        
+        db.session.commit()
+        
+        return jsonify({"task": task.to_json()}),200
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+@goals_bp.route("", methods=["POST", "GET"], strict_slashes=False)
+def goals():
+
+    if request.method == "GET":  
+        goals = Goal.query.all()
+        goals_response = []
+        for goal in goals: 
+        
+            goals_response.append(goal.to_json())
+            
+                
+        return jsonify(goals_response)
+    # using the "PUT" to add a new goal
+    else:
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return make_response(jsonify({"details": "Invalid data"}), 400)
+            
+        goal = Goal(title = request_body["title"])
+
+        
+        
+
+        db.session.add(goal)
+        db.session.commit()
+
+        return jsonify({"goal": goal.to_json()}),201
+    
+    
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"], strict_slashes=False)
+def handle_goal(goal_id):
+    # Try to find the goal with the given id
+
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return jsonify({"goal": goal.to_json()}),200
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+        return jsonify({"goal": goal.to_json()})
+
+    
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response({"details": (f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted")})    
+    
+    return {
+        "message": f"Goal with id {goal_id} was not found",
+        "success": False,
+    }, 404
+    
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_goals_task(goal_id):
+
+    form_data = request.get_json()
+
+    for task_id in form_data["task_ids"]: 
+        task = Task.query.get(task_id)
+        task.goal_id = goal_id
+
+    db.session.commit() 
+
+    return jsonify({"id": int(goal_id), "task_ids": form_data["task_ids"]}),200
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"], strict_slashes=False)
+def get_goals_task(goal_id):
```

Comment:
Small suggestion, add a blank line between route definitions

```suggestion

@goals_bp.route("/<goal_id>/tasks", methods=["GET"], strict_slashes=False)
def get_goals_task(goal_id):
```